src/relations/postgresql_provider.py

```diff
@@ -84,6 +118,22 @@ def _on_database_requested(self, event: DatabaseRequestedEvent) -> None:
             event.defer()
             return
 
+        user = None
+        password = None
+        try:
+            if requested_entities := event.requested_entity_secret_content:
+                for key, val in requested_entities.items():
+                    user = key
+                    password = val
+                    break
+                if user in SYSTEM_USERS:
+                    self.charm.unit.status = BlockedStatus(FORBIDDEN_USER_MSG)
+                    return
```

Comment:
Block if a system user has been requested.

Reply:
Should we block if the same username is requested by two relations?

Reply:
Updated the check to for existing users as well.